dkg-gadget/src/async_protocols/keygen/handler.rs

```diff
@@ -49,19 +49,24 @@ where
 				reason: "execute called twice with the same AsyncProtocol Parameters".to_string(),
 			})?;
 
+		let logger0 = params.logger.clone();
```

Comment:
Rust's borrowing rules. The logger needs to be cloned here in order to be moved into different closures. I didn't think I'd have to document this because the purpose of its existence is to appease the rust borrow checker, which is routine.